clang/lib/DPCT/AnalysisInfo.cpp

```diff
@@ -733,10 +733,14 @@ void DpctFileInfo::buildLinesInfo() {
     return;
   auto &SM = DpctGlobalInfo::getSourceManager();
 
-  auto FE = SM.getFileManager().getFile(FilePath);
-  if (std::error_code ec = FE.getError())
+  llvm::Expected<FileEntryRef> Result =
+      SM.getFileManager().getFileRef(FilePath);
+
+  if (auto E = Result.takeError()) {
     return;
-  auto FID = SM.getOrCreateFileID(FE.get(), SrcMgr::C_User);
+  }
+
+  auto FID = SM.getOrCreateFileID(Result.get(), SrcMgr::C_User);
```

Comment:
`auto FID = SM.getOrCreateFileID(*Result, SrcMgr::C_User);` ?

Reply:
OKay to me